crates/optimism/flashblocks/src/service.rs

```diff
@@ -167,6 +172,22 @@ where
             },
         )))
     }
+
+    /// Takes out `current` [`PendingBlock`] if `state` is not preceding it.
+    fn on_new_tip<E: Display>(
+        &mut self,
+        state: Result<CanonStateNotification<N>, E>,
```

Comment:
we can fully ignore the error case here because this stream will never lag and even if, we'll just skip to newest item